drone_control_gym.py

```diff
@@ -171,28 +192,25 @@ def step(self, action):
         mujoco.mj_step(self.model, self.drone)
         self.step_count += 1
 
-        self._update_drone_pose_from_sim()
-        self.drone_motor_thrust = list(self.drone.actuator_force)
+        self._update_drone_data_from_sim()
+        self.drone_motor_thrust = np.array(self.drone.actuator_force)
+        ic(self.drone_motor_thrust)
+        ic(self.drone_pose)
+        ic(self.imu_acc_data)
+        ic(self.imu_gyro_data)
 
-        print(f"quadcopter xyz: {self.drone_pose}")
-        print(f"quadcopter rpy: {self.drone_rpy}")
-
-        self.goal_attributes = self._calculate_goal_attributes()  # return list of [dx, dy, dz, d]
-        self.sensor_attributes = self._update_imu_data() # return array of [[accx, accy, accz], [velr, velp, vely]]
-        self.has_finished, self.reward = self._calculate_reward()  # return bool and float
-        
-        self.drone_last_ctrl = copy.deepcopy(self.drone.ctrl)
-        self.last_drone_pose = copy.deepcopy(self.drone_pose)
+        self.goal_attributes = (self._calculate_goal_attributes())  # return list of [dx, dy, dz, d]
+        self.has_finished, self.reward = (self._calculate_reward())  # return bool and float
 
         # eg 2.0, True, [0.3, 0.4, 0.1, 5.0], [33.0, 44.0, 55.0], [3.5, 3.5, 3.6, 3.7]
-        return self.reward, self.has_finished, self.goal_attributes, self.drone_rpy, self.drone_motor_thrust
+        return (self.reward, self.has_finished, self.goal_attributes, self.drone_rpy, self.last_drone_motor_thrust, self.drone_motor_thrust, self.last_imu_acc_data, self.imu_acc_data, self.last_imu_gyro_data, self.imu_gyro_data)
```

Comment:
why is this return a tuple? does the learning agent need the last motor thrust, last acc and last gyro data?